src/CMakeLists.txt

```diff
@@ -58,6 +58,13 @@ add_feature_info(futimens GIT_USE_FUTIMENS "futimens support")
 
 # qsort
 
+# for these tests, temporarily save CMAKE_C_FLAGS and disable warnings about
+# unused functions and parameters, otherwise they will always fail if
+# ENABLE_WERROR is on
+set(SAVED_CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+disable_warnings(unused-function)
+disable_warnings(unused-parameter)
```

Comment:
Clever! Because I think that we should switch to `check_prototype_definition` in a few other places in the codebase, could we abstract this out a bit?

Instead of putting the `disable_warnings` here in `CMakeLIsts` directly, can we create a function that disables the warnings and invokes `check_prototype_definition`, and then use that here instead?